store/details_v2.go

```diff
@@ -87,6 +87,139 @@ type storeSnapMedia struct {
 	Height int64  `json:"height"`
 }
 
+// storeInfoChannel is the channel description included in info results
+type storeInfoChannel struct {
+	Architecture string `json:"architecture"`
+	Name         string `json:"name"`
+	Risk         string `json:"risk"`
+	Track        string `json:"track"`
+}
+
+// storeInfoChannelSnap is the snap-in-a-channel of which the channel map is made
+type storeInfoChannelSnap struct {
+	storeSnap
+	Channel storeInfoChannel `json:"channel"`
+}
+
+// storeInfo is the result of v2/info calls
+type storeInfo struct {
+	ChannelMap []*storeInfoChannelSnap `json:"channel-map"`
+	Snap       storeSnap               `json:"snap"`
+	Name       string                  `json:"name"`
+	SnapID     string                  `json:"snap-id"`
+}
+
+func infoFromStoreInfo(si *storeInfo) (*snap.Info, error) {
+	if len(si.ChannelMap) == 0 {
+		// if a snap has no released revisions, it _could_ be returned
+		// (currently no, but spec is purposely ambiguous)
+		// we treat it as a 'not found' for now at least
+		return nil, ErrSnapNotFound
+	}
+
+	thisOne := si.ChannelMap[0]
```

Comment:
nitpick,  but I would prefer we made a copy here first:

thisSnap := si.ChannelMap[0].storeSnap